zip-0315.rst

```diff
@@ -1,9 +1,620 @@
+  balance;
+* if the wallet is not synchronized, the confirmed-spendable balance
+  will be zero.
+
+If auto-shielding is disabled, the wallet MAY report shielded and
+transparent balances separately. If it does so, it MUST make clear
+whether each reported balance corresponds to a confirmed-spendable,
+pending, or total subset of funds.
+
+Rationale for reporting of balances
+'''''''''''''''''''''''''''''''''''
+
+If auto-shielding is disabled, then separate shielded and transparent
+balances (and potentially, for expert users, separate shielded balances
+per pool) can constitute useful information. If auto-shielding is enabled
+then the wallet can and will automatically spend transparent TXOs in
+order to shield them, and so transparent TXOs need to be presented as
+pending, not as part of the balance spendable by the user.
+
+TODO: The specification of balance reporting is intended to give the user
+visibility into the operation of auto-shielding, opportunistic shielding,
+and pool migration/usage. (Does the spec satisfy this?)
+
+Reporting of transactions
+-------------------------
+
+If a transaction contains both spendable and watch-only TXOs, its spendable
+and watch-only incoming balances MUST be reported separately.
+
+Incoming transactions
+'''''''''''''''''''''
+
+A transaction is incoming if it contains unconfirmed-spendable TXOs.
+Incoming transactions SHOULD be reported with their number of confirmations
+and their balances as described in `Reporting of balances`_.
+
+Sent transactions
+'''''''''''''''''
+
+A transaction is sent if it was either:
+* created by the wallet, or
+* detected using the wallet's outgoing viewing keys, or
+* detected as having been sent from one of the wallet's watch-only addresses.
+
+Sent transactions SHOULD be reported with their number of confirmations,
+how long until they expire, and their balances as described in
+`Reporting of balances`_.
+
+
+Transaction creation
+--------------------
+
+Obtaining user consent for information leakage
+''''''''''''''''''''''''''''''''''''''''''''''
+
+Information leakage analysis
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The privacy provided by a Zcash transaction depends on the information leaked
+in the creation of that transaction and the process of it being broadcast for
+inclusion in the block chain.
+
+The requirements in this section are intended to minimize the leakage of such
+information where possible, and to ensure that the user is informed of any
+remaining information that would be leaked, and consents to such leakage.
+
+The list below describes the kinds of information that might be leaked. After
+a candidate transaction has been created and prior to it being revealed
+outside a trusted path to the user, the wallet user interface SHOULD obtain
+the user's consent for all of the leaked information.
+
+Assumption: There is always a transaction confirmation step for transactions
+that send funds out of the wallet.
+
```

Comment:
We will need to add some language here about the information leaked to the lightwallet server. Or, perhaps a separate section?

Reply:
The ZIP doesn't specifically refer to wallet types. perhaps it should?

from https://dl.acm.org/doi/abs/10.5555/3631672.3631676 there 4 broad types of wallets based on who is in custody of keys (sovereign or delegated) and how the blockchain is accessed (as a peer or through a trusted third party). 

<img width="902" alt="image" src="https://github.com/zcash/zips/assets/484008/2710aaca-409a-4ed0-8c2f-d25d8909ef97">

We might want to provide or reference definitions of types of wallets and scope the ones that we are referring to.